### PR TITLE
[s] blocks all teleport sources from infinite looping, reverts 24191

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

blocks looping
lets one shoot through perm teleporters effectively again / push multiple things through it at once

## Why It's Good For The Game

loops bad
teleporters teleporting multiple things good

## Testing

tried looping with all methods, emittered through permament with reflectors

## Changelog
:cl:
fix: Fixed infinite portal looping, permament teleporters can teleport 2 things at once again
/:cl:

### CODE REVIEW

File: code/game/machinery/teleporter.dm [hunk 1]
```diff
@@ -488,13 +498,15 @@
 		to_chat(AM, "You can't use this here.")
 		return
 
-	if(target && !recalibrating && !panel_open && !blockAI(AM))
+	if(target && !recalibrating && !panel_open && !blockAI(AM) && (MAX_ALLOWED_TELEPORTS_PER_PROCESS >= teleports_this_cycle))
```
Comment (DGamerL) [inline code/game/machinery/teleporter.dm:501]:
```suggestion
	if(target && !recalibrating && !panel_open && !blockAI(AM) && (teleports_this_cycle <= MAX_ALLOWED_TELEPORTS_PER_PROCESS ))
```

File: code/game/objects/effects/portals.dm [hunk 2]
```diff
@@ -132,6 +138,8 @@
 	return TRUE
 
 /obj/effect/portal/proc/attempt_teleport(atom/movable/victim, turf/destination, variance = 0, force_teleport = TRUE)
+	if(MAX_ALLOWED_TELEPORTS_PER_PROCESS <= teleports_this_cycle)
```
Comment (DGamerL) [inline code/game/objects/effects/portals.dm:141]:
```suggestion
	if(teleports_this_cycle >= MAX_ALLOWED_TELEPORTS_PER_PROCESS )
```

File: code/game/machinery/teleporter.dm [hunk 3]
```diff
@@ -488,13 +498,15 @@
 		to_chat(AM, "You can't use this here.")
 		return
 
-	if(target && !recalibrating && !panel_open && !blockAI(AM))
+	if(target && !recalibrating && !panel_open && !blockAI(AM) && (teleports_this_cycle <= MAX_ALLOWED_TELEPORTS_PER_PROCESS ))
```
Comment (DGamerL) [inline code/game/machinery/teleporter.dm:501]:
```suggestion
	if(target && !recalibrating && !panel_open && !blockAI(AM) && (teleports_this_cycle <= MAX_ALLOWED_TELEPORTS_PER_PROCESS))
```
Oopsie

File: code/game/objects/effects/portals.dm [hunk 4]
```diff
@@ -132,6 +138,8 @@
 	return TRUE
 
 /obj/effect/portal/proc/attempt_teleport(atom/movable/victim, turf/destination, variance = 0, force_teleport = TRUE)
+	if(teleports_this_cycle >= MAX_ALLOWED_TELEPORTS_PER_PROCESS )
```
Comment (DGamerL) [inline code/game/objects/effects/portals.dm:141]:
```suggestion
	if(teleports_this_cycle >= MAX_ALLOWED_TELEPORTS_PER_PROCESS)
```
Ghhh I hate myself

**Review by lewcc — APPROVED** [review]
e

**Pull request merged**